rswag/rswag
# Fix CSP header name based on Rack version

## Problem
Rack 3 now sends headers with lowercase names. This breaks the override CSP used by rswag, as the middleware writes the CSP with the name `Content-Security-Policy`, but Rails (with Rack 3) searches for `content-security-policy` to know if it should end its own policy.

## Solution
We configure the CSP header name based on the Rack version in the middleware

### This concerns this parts of the OpenAPI Specification:
It does not concern any part of the OpenAPI spec.

### The changes I made are compatible with:

### Related Issues
Related to #744

### Checklist
- [x] ~~Added documentation to README.md~~ Does not need documentation
- [ ] Added example of using the enhancement into [test-app](https://github.com/rswag/rswag/tree/master/test-app)

### Steps to Test or Reproduce
- Set up a Rails application using Rack 3.
- Set a CSP policy in Rails initializer
- Set up rswag in the application
- Try to display the webpage on rswag endpoint, page should be blank with CSP errors in the browser console.
